modules/fedimint-ln-client/src/lib.rs

```diff
@@ -916,6 +919,55 @@ impl LightningClientModule {
         Ok(())
     }
 
+    /// Continuously update the gateway cache whenever a gateway expires.
+    ///
+    /// Client integrators are expected to call this function in a spawned task.
+    pub async fn update_gateway_cache_continuously(&self, apply_meta: bool) -> ! {
+        let empty_gws_retry_builder = FibonacciBackoff::default()
+            .with_min_delay(Duration::from_secs(20))
+            .with_max_delay(Duration::from_secs(10 * 60))
+            .with_max_times(usize::MAX);
+        // retry immediately first time
+        let mut empty_gws_retries =
+            std::iter::once(Duration::ZERO).chain(empty_gws_retry_builder.build());
+
+        const ABOUT_TO_EXPIRE: Duration = Duration::from_secs(30);
+        loop {
+            let gateways = self.list_gateways().await;
+            // FIXME: filter gateways by vetted
+            let next_expiry = gateways
+                .into_iter()
+                .map(|x| x.ttl.saturating_sub(ABOUT_TO_EXPIRE))
+                .min();
+
+            let sleep_time = match next_expiry {
+                Some(next_expiry) => {
+                    // found non empty list => reset the retries iterator
+                    // try after 2 seconds first time
+                    empty_gws_retries = std::iter::once(Duration::from_secs(2))
+                        .chain(empty_gws_retry_builder.build());
```

Comment:
I guess we need this in case a gateway restarts, since it's likely to be back quickly, while some federations could be without one for longer and we don't want to keep polling at a high rate. Still don't like the mutability and complexity here, but don't see a more elegant way either :/

Reply:
I find it hard to figure out the exact strategy from the code itself, which is a problem for something that should be relatively simple. I would very much prefer something simpler here, which might be less optimal, at least for now and see if actually becomes a problem.

Reply:
We can still change this anytime so I want to avoid premature optimisation here.

Reply:
what would simple approach look like?

Reply:
in simple version, i would remove special case for empty gateways. then we will keep polling federation again and again which is horrible on mobile.

other way would be to have constant sleep, and constant sleep < 5-10 minutes is horrible on mobile.

Reply:
made it constant 10'